### PR TITLE
travis build and test failure fixes

### PR DESCRIPTION
**Pull request by pzhokhov**
Involved many changes - put actual latest version of the code, upgrade container ubuntu version to 16.04, fix the fake display setup (thanks @gdb !), fix some test failures.

### CODE REVIEW

File: gym/envs/robotics/robot_env.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def reset(self):
 
     def close(self):
         if self.viewer is not None:
-            self.viewer.finish()
+            # self.viewer.finish()
```
Comment (pzhokhov) [inline gym/envs/robotics/robot_env.py]: new mujoco MjViewer does not have method finish(). Leaving commented and not deleted for now, but we probably need to remove it entirely.

File: tox.ini [hunk 2]
```diff
@@ -17,8 +19,8 @@ deps =
     PyOpenGL
     pachi-py>=0.0.19
     box2d-py
-    doom_py>=0.0.11
-    mujoco_py<1.0.0,>=0.4.3
+    doom-py>=0.0.11
+    mujoco_py>=1.50.1,<1.50.2
```
Comment (pzhokhov) [inline tox.ini]: some gym code actually requires new version of mujoco and does not work with 0.5.x branch -> hence only python3 is tested. May need to test python3.6 separately.

File: .travis.yml [hunk 3]
```diff
@@ -3,24 +3,18 @@ language: python
...
   slack:
     secure: h/Mxm8K+avH/2W0818zCHmLloRPMFN4NJL01+VShvAkH80/acfjeq/+mMdWXXPL/oOB6kSHDk+GDhwR6+s03ZcPMn5INTFvFYqUc6UWmT+NXtOPxGTN0xda6MdYUkWQUKaMyjFrweZQOMOASFBIzPOq4XeVbM5aB8s4EJhnfAcYZhp/idwKbToVihN4KZgxlvZIFc8iEp1o9uSl5qrsaeYYYXRkb6mauacAwOo4/Chu+cOnoLUOnvhBFE3rV3doDNrbnoalO8XiExtgx5CIAYWrlMni7r2Q+LlzgwdyTH19ZtybPxJTZIIWSBQ2UtcoYdIEDcc36GcUwz1VUGg32mLJJnY2xw80CWR4ixFPpLwwP5Y99WTn8v094B4nmFTWOwNWXp3EkqtTN9XcJoRBqXB5ArucIPqrx57dOCljSKx22gL6WaF2p3stSAxIGFektGyGnisaELrFZG1C63aHoUPicj3gUlijmAoUmYaDRf6P1wnpXqBpKDAWWhAMSatvx1ekmEJgR7OQklQnnfjx9kENDUygNUWS4IQwN2qYieuzHFL3of7/30mTM43+Vt/vWN8GI7j01BXu6FNGGloHxjH1pt3bLP/+uj5BJsT2HWF+Z8XR4VE6cyVuKsQAFgCXwOkoDHALbcwsspONDIt/9ixkesgh1oFt4CzU3UuU5wYs=
     on_success: change
-  webhooks:
-    urls:
-      - https://hooks.zapier.com/hooks/catch/1711022/6ztmzh/
-      - https://hooks.zapier.com/hooks/catch/1711022/6zhc8p/
-    on_success: always
-    on_failure: always
+  # webhooks:
```
Comment (pzhokhov) [inline .travis.yml]: so what exactly do these webhooks do? When curling them, I get some generic JSON success message (`{"status": "success", "attempt": "5af229fc-b676-4f1e-8436-1886aa2a3a10", "id": "4d7cd20f-76e2-47c4-a2ee-f742a35a0267", "request_id": "5af229fc-b676-4f1e-8436-1886aa2a3a10"}` 
Commenting out to see if anyone would cry of their absense :)

Comment (gdb) [reply .travis.yml]: Legacy, no issue with removing!

File: test.dockerfile [hunk 4]
```diff
@@ -1,44 +1,68 @@
...
+    pip install tox && \
+
+# Download mujoco
+    mkdir /root/.mujoco && \
+    cd /root/.mujoco  && \
+    wget https://www.roboti.us/download/mjpro150_linux.zip  && \
+    unzip mjpro150_linux.zip && \
+
+# Inject the mujoco license key (if planning to publish image, this step has to be runtime)
+    echo $MUJOCO_KEY | base64 --decode > mjkey.txt && \
```
Comment (gdb) [inline test.dockerfile]: Why have this in the Dockerfile at all? It's not harmful per-se, but might be nicer to just keep `MUJOCO_KEY` to the "docker run" invocation so we don't have to think about it.

Comment (pzhokhov) [reply test.dockerfile]: sounds good, I'll adjust it

File: .travis.yml (hunk 3 above)
Comment (pzhokhov) [reply .travis.yml]: okay, will remove

File: bin/docker_entrypoint [hunk 5]
```diff
@@ -2,6 +2,10 @@
 # This script is the entrypoint for our Docker image.
 
 set -e
+
+# Inject the mujoco license key (if planning to publish image, this step has to be runtime)
```
Comment (christopherhesse) [inline bin/docker_entrypoint]: the step is runtime now right?

Comment (pzhokhov) [reply bin/docker_entrypoint]: correct

**Pull request merged**